pkg/kubectl/cmd/apply.go

```diff
@@ -684,6 +693,22 @@ type patcher struct {
 }
 
 func (p *patcher) patchSimple(obj runtime.Object, modified []byte, source, namespace, name string, errOut io.Writer) ([]byte, runtime.Object, error) {
+	if p.serverDryRun {
+		doc, err := p.discoveryClient.OpenAPISchema()
```

Comment:
patchSimple is already a large function. Should we break this check into it's own patcher method? Additionally, can we add significant comments to describe why this is necessary (what bug this fixes).